util/pkg/vfs/gsfs.go

```diff
@@ -374,6 +386,59 @@ func (p *GSPath) Hash(a hashing.HashAlgorithm) (*hashing.Hash, error) {
 	return &hashing.Hash{Algorithm: hashing.HashAlgorithmMD5, HashValue: md5Bytes}, nil
 }
 
+type terraformGSObject struct {
+	Bucket   string                   `json:"bucket" cty:"bucket"`
+	Name     string                   `json:"name" cty:"name"`
+	Source   string                   `json:"source" cty:"source"`
+	Provider *terraformWriter.Literal `json:"provider,omitempty" cty:"provider"`
+}
+
+type terraformGSObjectAccessControl struct {
+	Bucket     string                   `json:"bucket" cty:"bucket"`
+	Object     *terraformWriter.Literal `json:"object" cty:"object"`
+	RoleEntity []string                 `json:"role_entity" cty:"role_entity"`
+	Provider   *terraformWriter.Literal `json:"provider,omitempty" cty:"provider"`
+}
+
+func (p *GSPath) RenderTerraform(w *terraformWriter.TerraformWriter, name string, data io.Reader, acl ACL) error {
+	bytes, err := ioutil.ReadAll(data)
+	if err != nil {
+		return fmt.Errorf("reading data: %v", err)
```

Comment:
Nit (no need to fix): I've started using %w in new code.  I realize this might be copied-and-pasted from somewhere though!

Eventually I suspect we'll just bulk-replace, so I don't think it's worth fixing here.